vernac/declare.mli

```diff
@@ -169,7 +150,7 @@ type import_status = Locality.import_status = ImportDefaultBehavior | ImportNeed
   internal specify if the constant has been created by the kernel or by the
   user, and in the former case, if its errors should be silent
 
-  XXX: Scheduled for removal from public API, use `DeclareDef` instead *)
+  XXX: Scheduled for removal from public API, use `declare_definition` instead *)
 val declare_constant
```

Comment:
I don't think it's bad to have a way to declare using the kernel's language directly.
For instance the lean importer doesn't want to restrict universes and more generally doesn't want to mess with evar maps.

Reply:
Good point, there is tension on how to prevent most users from using lower-level APIs vs still allowing those that need them use them. There are several approaches we could think of implementing:
- requiring users of low-level APIs to use kernel entries directly [after we tweak state a bit]
- providing some optional parameters to the high-level
- ??
for now I will modify the comment to something way softer.

Reply:
Comments modified; the API in this file still needs a lot of work IMO, once 8.13 is open I will submit quite a few cleanups; I will likely wrap the low-level stuff in a different way.

Reply:
FTR, this is precisely the kind of use I was advocating for in a previous comment.

Reply:
I see, I think that for now we can live with the `definition_entry` constructor (which could become smart once we handle side effects differently) and discuss a bit more later.

Reply:
Note that `Evd.merge_universe_subst` has become unused after this change; maybe should be removed?